server/resources/resource_management.go

```diff
@@ -851,9 +851,33 @@ func DeleteResource(context middleware.Context,
 		}
 		policy.AddCustomFilters(filter, auth.TenantID())
 		resource, fetchErr = preTransaction.Fetch(resourceSchema, filter, nil)
+		if fetchErr != nil {
+			switch fetchErr {
+			case transaction.ErrResourceNotFound:
+				policy, err := loadPolicy(context, "read", strings.Replace(resourceSchema.GetSingleURL(), ":id", resourceID, 1), auth)
+				if err != nil {
+					return err
+				}
+				context["policy"] = policy
```

Comment:
I'm not sure about this line, since context is not used in this function after that assignment.